OpenRA.Mods.Common/Widgets/Logic/Editor/SaveMapLogic.cs

```diff
@@ -237,41 +232,79 @@ public SaveDirectory(Folder folder, string displayName, MapClassification classi
 					return;
 
 				var combinedPath = Platform.ResolvePath(Path.Combine(selectedDirectory.Folder.Name, filename.Text + fileTypes[fileType].Extension));
+				SaveMap(modData, map, combinedPath, saveMap);
+			};
+		}
 
-				if (map.Package?.Name != combinedPath)
+		public static void SaveMap(ModData modData, Map map, string combinedPath, Action<string> saveMap)
+		{
+			if (map.Package?.Name != combinedPath)
+			{
+				// When creating a new map or when file paths don't match
+				if (modData.MapCache.Any(m => m.Status == MapStatus.Available && m.Package?.Name == combinedPath))
 				{
-					// When creating a new map or when file paths don't match
-					if (modData.MapCache.Any(m => m.Status == MapStatus.Available && m.Package?.Name == combinedPath))
-					{
-						ConfirmationDialogs.ButtonPrompt(modData,
-							title: OverwriteMapFailedTitle,
-							text: OverwriteMapFailedPrompt,
-							confirmText: SaveMapFailedConfirm,
-							onConfirm: () => saveMap(combinedPath),
-							onCancel: () => { });
-
-						return;
-					}
+					ConfirmationDialogs.ButtonPrompt(modData,
+						title: OverwriteMapFailedTitle,
+						text: OverwriteMapFailedPrompt,
+						confirmText: SaveMapFailedConfirm,
+						onConfirm: () => saveMap(combinedPath),
+						onCancel: () => { });
+
+					return;
 				}
-				else
+			}
+			else
+			{
+				// When file paths match
+				var recentUid = modData.MapCache.GetUpdatedMap(map.Uid);
+				if (recentUid != null && map.Uid != recentUid && modData.MapCache[recentUid].Status == MapStatus.Available)
 				{
-					// When file paths match
-					var recentUid = modData.MapCache.GetUpdatedMap(map.Uid);
-					if (recentUid != null && map.Uid != recentUid && modData.MapCache[recentUid].Status == MapStatus.Available)
-					{
-						ConfirmationDialogs.ButtonPrompt(modData,
-							title: OverwriteMapOutsideEditTitle,
-							text: OverwriteMapOutsideEditPrompt,
-							confirmText: SaveMapMapOutsideConfirm,
-							onConfirm: () => saveMap(combinedPath),
-							onCancel: () => { });
-
-						return;
-					}
+					ConfirmationDialogs.ButtonPrompt(modData,
+						title: OverwriteMapOutsideEditTitle,
+						text: OverwriteMapOutsideEditPrompt,
+						confirmText: SaveMapMapOutsideConfirm,
+						onConfirm: () => saveMap(combinedPath),
+						onCancel: () => { });
+
+					return;
 				}
+			}
 
-				saveMap(combinedPath);
-			};
+			saveMap(combinedPath);
+		}
+
+		public static void SaveMapInner(Map map, IReadWritePackage package, World world, ModData modData)
+		{
+			map.RequiresMod = modData.Manifest.Id;
+
+			try
+			{
+				if (package == null)
+					throw new ArgumentNullException(nameof(package));
```

Comment:
It makes no sense to crash and lose the map progress. Instead we should just add transient notification that the save failed or open SaveMapLogic menu